src/ios/CameraSessionManager.m

```diff
@@ -109,6 +119,20 @@ - (void) setupSession:(NSString *)defaultCamera completion:(void(^)(BOOL started
     }];
 }
 
+- (AVCaptureSessionPreset) calculateResolution:(NSInteger)targetSize {
+    if (targetSize >= 3840) {
+        return AVCaptureSessionPreset3840x2160;
+    } else if (targetSize >= 1920) {
+        return AVCaptureSessionPreset1920x1080;
+    } else if (targetSize >= 1280) {
+        return AVCaptureSessionPreset1280x720;
+    } else if (targetSize >= 640) {
+        return AVCaptureSessionPreset640x480;
+    } else {
+        return AVCaptureSessionPreset352x288;
+    }
```

Comment:
switch?

Reply:
@sc-nick Cannot use switch as we are unable to test range of values in Objective C

Reply:
ya I'm sorry I didn't check the test

Reply:
@zfir ya sorry I didn't check the test